src/rpc/client.cpp

```diff
@@ -214,15 +214,25 @@ CRPCConvertTable::CRPCConvertTable()
 
 static CRPCConvertTable rpcCvtTable;
 
+static bool IsHexDigit(const char c) {
+    return ('0' <= c && c <= '9') || ('a' <= c && c <= 'f') || ('A' <= c && c <= 'F');
+}
+
+static bool IsHashVal(const std::string& str) {
+    return (str.length() == 64) && std::all_of(str.begin(), str.end(), IsHexDigit);
```

Comment:
Could use bool `IsHex(const std::string&)` from src/util/strencodings.h.

Reply:
Hm, my impression was that client.h/cpp should be stand-alone and not depend on other parts of the project? Maybe I am wrong about that but I thought I heard that somewhere and looking at the includes seemed to confirm it. Otherwise, I guess I could also just use `ParseHashV` from rpc/util.cpp?

Reply:
AFAIK it can't include server side components.

Reply:
Makes sense, changed to use `IsHex`.